WalletWasabi.Backend/Controllers/BlockchainController.cs

```diff
@@ -416,4 +416,88 @@ private async Task<StatusResponse> FetchStatusAsync(CancellationToken cancellati
+		var cacheOptions = new MemoryCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(3) };
+
+		return await Cache.GetCachedResponseAsync(
+			cacheKey,
+			action: (string request, CancellationToken token) => GetUnconfirmedTransactionChainNoCacheAsync(txId, token),
+			options: cacheOptions,
+			cancellationToken);
+	}
+
+	private async Task<List<UnconfirmedTransactionChainItem>> GetUnconfirmedTransactionChainNoCacheAsync(uint256 txId, CancellationToken cancellationToken)
+	{
+		Dictionary<uint256, Transaction> transactionsLocalCache = new();
+		Dictionary<uint256, UnconfirmedTransactionChainItem> unconfirmedTxsChainById = new();
+		var mempool = Global.HostedServices.Get<MempoolMirror>();
+
+		// TODO: Use Transaction cache.
+		var requestedTransaction = await RpcClient.GetRawTransactionAsync(txId, true, cancellationToken);
+
+		List<Transaction> toFetchFeeList = new() { requestedTransaction };
+
+		while (toFetchFeeList.Count > 0)
+		{
+			List<Coin> inputs = new();
+			HashSet<Transaction> parentTxs = new();
+
+			var currentTx = toFetchFeeList.First();
+			foreach (var input in currentTx.Inputs)
+			{
+				if (!transactionsLocalCache.TryGetValue(input.PrevOut.Hash, out var parentTx))
+				{
+					parentTx = await RpcClient.GetRawTransactionAsync(input.PrevOut.Hash, true, cancellationToken);
+					transactionsLocalCache.Add(input.PrevOut.Hash, parentTx);
+				}
+
+				parentTxs.Add(parentTx);
+				TxOut txOut = parentTx.Outputs[input.PrevOut.N];
+				inputs.Add(new Coin(input.PrevOut, txOut));
+			}
+
+			var unconfirmedChildrenTxs = mempool
+				.GetSpenderTransactions(currentTx.Outputs.Select((txo, index) => new OutPoint(currentTx, index)))
+				.ToHashSet();
+
+			// Add the children to the local transaction cache.
+			foreach (var childTx in unconfirmedChildrenTxs)
+			{
+				if (!transactionsLocalCache.ContainsKey(childTx.GetHash()))
+				{
+					transactionsLocalCache.Add(childTx.GetHash(), childTx);
+				}
+			}
+
+			// Remove the item we worked on.
+			toFetchFeeList.Remove(currentTx);
+
+			var unconfirmedParents = parentTxs.Where(x =>
+				mempool.GetMempoolHashes().Contains(x.GetHash()));
+
+			// Fee and size of all unconfirmed parents and children not already known are required to calculate the effective fee rate of the current transaction.
+			toFetchFeeList.AddRange(
+				unconfirmedParents.Where(x => !unconfirmedTxsChainById.ContainsKey(x.GetHash()))
+					.Union(unconfirmedChildrenTxs.Where(x => !unconfirmedTxsChainById.ContainsKey(x.GetHash()))));
+
+			unconfirmedTxsChainById.Add(
+				currentTx.GetHash(),
+				new UnconfirmedTransactionChainItem(
+					TxId: currentTx.GetHash().ToString(),
+					Size: currentTx.GetVirtualSize(),
+					Fee: currentTx.GetFee(inputs.ToArray()),
+					Parents: unconfirmedParents.Select(x => x.GetHash().ToString()).ToHashSet(),
+					Children: unconfirmedChildrenTxs.Select(x => x.GetHash().ToString()).ToHashSet()));
+		}
```

Comment:
Where is the depth limit threshold?

Reply:
Added a naive way for the limitation. [8877998](https://github.com/zkSNACKs/WalletWasabi/pull/12302/commits/88779986996ba05170df88224fb120cfd3b3d37c)